thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/alert/feed/UnionAnomalyFeed.java

```diff
@@ -0,0 +1,106 @@
+
+  private List<AnomalyFetcher> anomalyFetchers;
+  private List<AlertFilter> alertFilters;
+  private AlertSnapshotDTO alertSnapshot;
+  private DateTime alertTime;
+
+  public UnionAnomalyFeed(){
+    this.anomalyFetchers = new ArrayList<>();
+    this.alertFilters = new ArrayList<>();
+    this.alertTime = DateTime.now();
+  }
+
+  @Override
+  public void init(AlertFilterFactory alertFilterFactory, AnomalyFeedConfig anomalyFeedConfig) {
+    this.alertSnapshotDAO = DAORegistry.getInstance().getAlertSnapshotDAO();
+    this.alertFilterFactory = alertFilterFactory;
+    AnomalySource anomalySourceType = anomalyFeedConfig.getAnomalySourceType();
+    String anomalySource = anomalyFeedConfig.getAnomalySource();
+    if (anomalyFeedConfig.getAlertSnapshotId() != null) {
+      alertSnapshot = alertSnapshotDAO.findById(anomalyFeedConfig.getAlertSnapshotId());
+    }
+    List<AnomalyFetcherConfig> anomalyFetcherSpecs = anomalyFeedConfig.getAnomalyFetcherConfigs();
+    List<Map<String, String>> alertFilterSpecs = anomalyFeedConfig.getAlertFilterConfigs();
+
+    for (AnomalyFetcherConfig anomalyFetcherSpec : anomalyFetcherSpecs) {
+      AnomalyFetcher anomalyFetcher;
+      try {
+        anomalyFetcher = AnomalyFetcherFactory.fromClassName(anomalyFetcherSpec.getType());
+      } catch (Exception e) {
+        LOG.error("Cannot instantiate anomaly fetcher {}", anomalyFetcherSpec.getType(), e);
+        continue;
+      }
+
+      anomalyFetcherSpec.setAnomalySourceType(anomalySourceType);
+      anomalyFetcherSpec.setAnomalySource(anomalySource);
+      anomalyFetcher.init(anomalyFetcherSpec);
+      anomalyFetchers.add(anomalyFetcher);
+    }
+
+    for(Map<String, String> alertFilterSpec : alertFilterSpecs) {
+      alertFilters.add(alertFilterFactory.fromSpec(alertFilterSpec));
+    }
+  }
+
+  @Override
+  public Collection<MergedAnomalyResultDTO> getAnomalyFeed() {
+    Set<MergedAnomalyResultDTO> mergedAnomalyResultSet = new HashSet<>();
+    for (AnomalyFetcher anomalyFetcher : anomalyFetchers) {
+      mergedAnomalyResultSet.addAll(anomalyFetcher.getAlertCandidates(alertTime, alertSnapshot));
+    }
+    List<MergedAnomalyResultDTO> mergedAnomalyResults = new ArrayList<>(mergedAnomalyResultSet);
+    AlertFilterHelper.applyFiltrationRule(mergedAnomalyResults, alertFilterFactory);
+
+    for(AlertFilter alertFilter : alertFilters) {
+      Iterator<MergedAnomalyResultDTO> mergedAnomalyIterator = mergedAnomalyResults.iterator();
+      while (mergedAnomalyIterator.hasNext()) {
+        MergedAnomalyResultDTO mergedAnomalyResult = mergedAnomalyIterator.next();
+        if (!alertFilter.isQualified(mergedAnomalyResult)) {
+          mergedAnomalyIterator.remove();
+        }
+      }
+    }
+    return mergedAnomalyResults;
+  }
+
+  @Override
+  public void updateSnapshot(Collection<MergedAnomalyResultDTO> alertedAnomalies) {
+    if (alertSnapshot == null) {
+      LOG.warn("No alertSnapshot was initiated.");
```

Comment:
Throw an exception or handle null pointer.

Assuming that alertSnapshot might not be initialized in init in the first place, handling the null pointer is probably a better option.

Reply:
It has fixed. Thanks for the comments.